makefiles/features_precheck.inc.mk

```diff
@@ -0,0 +1,17 @@
+# Check if provided features are available
+
+ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)) $(filter newlib,$(FEATURES_PROVIDED)))
+  ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)))
+    # Test if picolibc.specs is available
+    ifneq ($(shell $(LINK) -specs=picolibc.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+      FEATURES_PROVIDED := $(filter-out picolibc,$(FEATURES_PROVIDED))
```

Comment:
This is not declarative, FEATURES so far are declarative.

Reply:
I think: I fixed that with blacklisting